validator/sawtooth_validator/state/state_delta_processor.py

```diff
@@ -119,6 +119,25 @@ def add_subscriber(
         LOGGER.debug('Added Subscriber %s for %s',
                      connection_id, address_prefix_filters)
 
+        if known_block_id != self._block_store.chain_head.identifier:
+            LOGGER.debug('Catching up Subscriber %s from %s to %s',
+                         connection_id,
+                         known_block_id,
+                         self._block_store.chain_head.identifier[:8])
+            catch_up_blocks = []
+
+            for block in self._block_store.get_predecessor_iter():
```

Comment:
It looks like the _BlockPredecessorIterator will throw a stop iteration if the block store has changed status, like having switched forks and the predecessor is no longer in the BlockStore. How is that detected and handled here?